packages/mongodb-memory-server-core/src/MongoMemoryServer.ts

```diff
@@ -201,20 +251,37 @@ export default class MongoMemoryServer {
     return uri;
   }
 
+  /**
+   * Get a mongodb-URI for a different DataBase
+   * @param otherDbName Set this to "true" to generate a random DataBase name, otherwise a string to specify a DataBase name
+   */
   async getConnectionString(otherDbName: string | boolean = false): Promise<string> {
+    // should this function be marked deprecated? because it is just a pass-through to getUri
     return this.getUri(otherDbName);
   }
 
+  /**
+   * Get the Port of the currently running Instance
+   * Note: calls "ensureInstance"
+   */
   async getPort(): Promise<number> {
     const { port }: MongoInstanceDataT = await this.ensureInstance();
```

Comment:
Do you really want to call `ensureInstace` here, instead of returning `false`/`null` if not running?

Reply:
Yep. The following code, quite simple for testing purposes
```
const mongod = new MongoMemoryServer();
const uri = await mongod.getUri();
```
rather than
```
const mongod = new MongoMemoryServer();
await mongod.start();
const uri = mongod.getUri();
```

Reply:
For now, it will be quite difficult to change behavior and force developers rewrite existing code. It will need strong motivation for changing it and publishing a new major version.

Reply:
i know that it is easier for testing, but i just thought that not everyone wants the instance started if it is *somehow* not started

Reply:
Yep for such uncommon scenario was added `autoStart: false` option.